merci/evaluate.py

```diff
@@ -114,3 +115,46 @@ def evaluate(self) -> float:
 
         score = merci.measures.reliability_estimation(y_test, y_transductive)
         return score
+
+
+class DistributionShiftEvaluator(ModelEvaluator):
+    """
+    Evaluate robustness to distribution shift
+    """
+    def __init__(
+        self,
+        model: merci.types.Classifier,
+        train_dataset: merci.types.Dataset,
+        test_dataset: merci.types.Dataset,
+    ) -> None:
+        """
+        Initialize the transductive model evaluator.
+
+        :param model: Classification model to evaluate
+        :param train_dataset: A tuple containing input and target data for training
+        :param test_dataset: A tuple containing input and target data for testing
+        """
+        super().__init__(model, train_dataset, test_dataset)
+        self.thresholds = [0.75, 0.5, 0.25, 0.1]
+
+    def evaluate(self) -> float:
+        """
+        Evaluate the model using the provided dataset.
+
+        :return: The distribution shift robustness estimation of the model on the dataset
+        """
+        x_train, y_train = self.train_dataset
+        x_test, y_test = self.test_dataset
+        self.model.fit(x_train, y_train)
+        scores = []
+        class0 = np.where(y_train == 0)[0]
+        class1 = np.where(y_train == 1)[0]
+        for cls, other_cls in zip([class0, class1], [class1, class0]):
+            for thr in self.thresholds:
+                reduced = np.random.choice(cls, np.floor(len(cls)*thr).astype(int))
```

Comment:
If we use a random method here, imo for reproducibility we should add a `seed` parameter to the evaluator